examples/demo-stf/src/sov-cli/native.rs

```
@@ -0,0 +1,487 @@
+        module_name: &str,
+        call_data_path: P,
+    ) -> anyhow::Result<Vec<u8>> {
+        let call_data = std::fs::read_to_string(&call_data_path).with_context(|| {
+            format!(
+                "Failed to read call data from {:?}",
+                call_data_path.as_ref()
+            )
+        })?;
+        cmd_parser(module_name, &call_data)
+    }
+}
+#[cfg(feature = "native")]
+pub fn main() {
+    let cli = Cli::parse();
+    match cli.command {
+        Commands::SerializeCall {
+            sender_priv_key_path,
+            module_name,
+            call_data_path,
+            nonce,
+        } => {
+            let serialized =
+                SerializedTx::new(&sender_priv_key_path, &module_name, &call_data_path, nonce)
+                    .unwrap_or_else(|e| panic!("Call message serialization error: {}", e));
+
+            let mut bin_path = PathBuf::from(call_data_path);
+            bin_path.set_extension("dat");
+
+            let mut file = File::create(bin_path)
+                .unwrap_or_else(|e| panic!("Unable to crate .dat file: {}", e));
+
+            let raw_contents = hex::encode(serialized.raw.data).as_bytes().to_vec();
+            file.write_all(&raw_contents)
+                .unwrap_or_else(|e| panic!("Unable to save .dat file: {}", e));
+        }
+        Commands::MakeBlob { path_list } => {
+            let mut data_list: Vec<Vec<u8>> = vec![];
+
+            for path in path_list {
+                let mut f = File::open(path).expect("Unable to open file");
+                let mut hex_string = String::new();
+                f.read_to_string(&mut hex_string)
+                    .expect("Unable to read the file");
+
+                // decode the hex string to bytes
+                let bytes = hex::decode(hex_string.trim()).expect("Decoding failed");
+
+                // push it into data_list
+                data_list.push(bytes);
+            }
+            let blob = data_list.try_to_vec().unwrap();
```

nit: I think this panic is ok (IIUC, an error should not actually be possible here), but lets add a comment documenting _why_ a panic is ok: the error is unexpected and unrecoverable, likely indicating a bug

---

Yes, it should never panic. It can only panic, if it fails to allocate vec. I think `Result` signature is leaked from borsh serializing into io.